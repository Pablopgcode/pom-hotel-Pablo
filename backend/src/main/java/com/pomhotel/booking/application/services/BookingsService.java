package com.pomhotel.booking.application.services;

import com.pomhotel.booking.application.models.BookingsModel;

import java.sql.Date;
import java.util.List;

//--- Service Interface -------------------------------------------
public interface BookingsService {

    BookingsModel findById(long id);

    List<BookingsModel> findAll();

    long saveOrUpdate(BookingsModel model);

    void deleteById(long id);

    void delete(BookingsModel model);


    public List<Date> getReservedDates(long id);
}
