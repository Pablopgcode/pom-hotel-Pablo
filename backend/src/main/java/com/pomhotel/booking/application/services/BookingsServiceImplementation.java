package com.pomhotel.booking.application.services;

import com.pomhotel.booking.application.domain.entities.BookingsEntity;
import com.pomhotel.booking.application.domain.factories.BookingsFactory;
import com.pomhotel.booking.application.models.BookingsModel;
import com.pomhotel.booking.application.repositories.BookingsRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;
import java.util.stream.Collectors;

//--- Service ----------------------------------------------------------
@Service
public class BookingsServiceImplementation implements BookingsService{

    //--- Repositories & Factories needed ------------------------------
    BookingsRepository repository;
    BookingsFactory factory;

    //--- Constructor --------------------------------------------------
    @Autowired
    public BookingsServiceImplementation(BookingsRepository repository, BookingsFactory factory) {
        this.repository = repository;
        this.factory = factory;
    }

    //--- Functions ----------------------------------------------------
    @Override
    public BookingsModel findById(long id) {
        return factory.createModel(repository.findById(id));
    }

    @Override
    public List<BookingsModel> findAll() {
        List<BookingsEntity> entities = repository.findAll();
        List<BookingsModel> models = entities.stream().map(entity -> {
            return factory.createModel(entity);
        }).collect(Collectors.toList());
        return models;
    }

    @Override
    public long saveOrUpdate(BookingsModel model) {
        long id = repository.saveOrUpdate(factory.createEntity(model));
        return id;
    }

     @Override
     public List<Date> getReservedDates(long id) {
         List<Object[]> entities = repository.getReservedDates(id);
         List<Date> reservedDates = new ArrayList<>();
         for (int i=0 ; i < entities.size(); i++){
            Date checkIn = (Date) entities.get(i)[0];
            Date checkOut = (Date) entities.get(i)[1];
            reservedDates.add(checkIn);
            Date newDate = checkIn;
            while (!newDate.equals(checkOut)) {
                newDate = new Date(newDate.getTime() + TimeUnit.DAYS.toMillis(1));
                reservedDates.add(newDate);
            }
         }
         return reservedDates;
    }

    @Override
    public void deleteById(long id) {
        repository.deleteById(id);
    }

    @Override
    public void delete(BookingsModel model) {
        repository.delete(factory.createEntity(model));
    }

}
