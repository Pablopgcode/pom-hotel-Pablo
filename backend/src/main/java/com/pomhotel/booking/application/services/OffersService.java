package com.pomhotel.booking.application.services;

import com.pomhotel.booking.application.models.OffersModel;
import com.pomhotel.booking.application.models.RoomtypesModel;

import java.util.List;

public interface OffersService {
    OffersModel findById(long id);

    List<OffersModel> findAll();

    void saveOrUpdate(OffersModel model);
}
