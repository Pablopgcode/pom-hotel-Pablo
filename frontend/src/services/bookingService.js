import axios from 'axios';
import { reduceEachLeadingCommentRange } from 'typescript';

const BOOKINGS_URL = "http://localhost:8080/boot";
class BookService{
    getBookings(){
        return axios.get( BOOKINGS_URL + '/booknow');
    }

    getTotalPrice(startDate, endDate, id){
        return axios.post(BOOKINGS_URL + '/getTotalPrice',{            
            checkIn: startDate,
            checkOut: endDate,
            roomId: id                      
        })
    }

    saveBooking(id, startDate, endDate, totalPrice, safebox, wedge, laundry, parking){
        return axios.post(BOOKINGS_URL + '/booknow',{
            roomId: id,
            checkIn: startDate,
            checkOut: endDate,
            totalPrice: totalPrice,
            safebox: safebox,
            wedge: wedge,
            laundry: parking
        })
    }
}

export default new BookService

