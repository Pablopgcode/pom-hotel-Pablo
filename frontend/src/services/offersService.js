import axios from 'axios';
const BOOT_URL = "http://localhost:8080/boot";

class OffersService{
    getOffers(){
        return axios.get( BOOT_URL + '/offers');
    }
}
export default new OffersService()
