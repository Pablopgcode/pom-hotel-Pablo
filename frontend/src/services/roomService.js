import axios from 'axios';

const ROOMS_URL = "http://localhost:8080/boot";
class RoomService{
    getRooms(){
        return axios.get( ROOMS_URL + '/rooms');
    }   
}
export default new RoomService()