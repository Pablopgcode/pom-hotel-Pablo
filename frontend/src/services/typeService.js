import axios from 'axios';

const BOOKINGS_URL = "http://localhost:8080/boot";
class TypeService{
    getTypes(){
        return axios.get( BOOKINGS_URL + '/types');
    }
}
export default new TypeService()