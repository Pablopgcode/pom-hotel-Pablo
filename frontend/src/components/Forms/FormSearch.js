import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import 'assets/css/bootstrap.min.css';
import '../../assets/css/various-ui-comp.css'
import TypeService from '../../services/typeService.js';
import {now} from '../../services/dateservice';
class FormSearch extends Component {
    /* Props */
    constructor(props) {
        super(props)
        this.state = {
            rtypes: [],
            reserved: [],
            guests: "1",
            startDate: now,
            endDate: now,
            type: "",
            minprice: "1",
            maxprice: "1000"
        }
    }
    /* obtain types of rooms */
    componentDidMount() {
        TypeService.getTypes().then((res) => {
            this.setState({ rtypes: res.data });
        });
    }

    /*Handlers*/
    handleGuests (event){
        this.setState({guests: event.target.value});
    }

    handleTypes (event){
        this.setState({type: event.target.value}); 
    }

    handleMinprice (event){
        this.setState({minprice: event.target.value});
    }
    
    handleMaxprice (event){
        this.setState({maxprice: event.target.value});
    }

    submitFormData(ev){
        ev.preventDefault();
        this.props.onFilterChange(this.state)
    }
    
    render() {
        return (
            <div className="col-md-3">
                <div className="sidebar-wrap img-raised" style={{ marginTop: '50px' }}>
                    <div className="side search-wrap animate-box">
                        <h2 className="sidebar-heading">Find your room</h2>
                        <form onSubmit={(e)=> this.submitFormData(e)} id="roomsSearch" className="colorlib-form">
                            <div className="row">
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="checkin">Check-in:</label>
                                        <div className="form-field">
                                            <DatePicker
                                                className="datum form-control"
                                                selected={ this.state.startDate}
                                                onChange={ (date) => this.setState({startDate: date}) }
                                                minDate={new Date()}
                                                name="startDate"
                                                dateFormat="dd/MM/yyyy"
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="checkout">Check-out:</label>
                                        <div className="form-field">
                                            <DatePicker
                                            className="datum form-control"
                                            selected={ this.state.endDate<this.state.startDate ? this.state.startDate : this.state.endDate || this.state.startDate}
                                            onChange={  (date) => this.setState({endDate: date})  }
                                            minDate={this.state.startDate}
                                            name="endDate"
                                            dateFormat="dd/MM/yyyy"
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="guests">Guest</label>
                                        <div className="form-field">
                                            <select id="guests" name="guests" className="form-control" value={this.state.guests} onChange={(e) => this.handleGuests(e)}>
                                                <option value="1">1</option>
                                                <option value="2">2</option>
                                                <option value="3">3</option>
                                                <option value="4">4</option>
                                                <option value="5">5+</option>
                                            </select>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="roomtypes">Room Types</label>
                                        <div className="form-field"> 
                                            <select name="type" id="roomtype" className="form-control" value={this.state.value} onChange={(e) => this.handleTypes(e)}>
                                                <option value="0">All types</option>
                                                {
                                                    this.state.rtypes.map(
                                                        type =>
                                                            <option value={type.id}>{type.name}</option>
                                                    )
                                                }
                                            </select>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="pricefrom">Price from:</label>
                                        <div className="form-field">
                                            <select name="minprice" id="pricefrom" className="form-control" value={this.state.minprice} onChange={(e) => this.handleMinprice(e)}>
                                                <option value="1" disabled={this.state.maxprice < 1 ? 'true' : ''}>1</option>
                                                <option value="50" disabled={this.state.maxprice < 50 ? 'true' : ''}>50</option>
                                                <option value="100" disabled={this.state.maxprice < 100 ? 'true' : ''}>100</option>
                                                <option value="200" disabled={this.state.maxprice < 200 ? 'true' : ''}>200</option>
                                                <option value="300" disabled={this.state.maxprice < 300 ? 'true' : ''}>300</option>
                                            </select>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-6">
                                    <div className="form-group">
                                        <label for="priceto">Price to:</label>
                                        <div className="form-field">
                                            <select name="maxprice" id="priceto" className="form-control" value={this.state.maxprice} onChange={(e) => this.handleMaxprice(e)}>
                                                <option value="100" disabled={this.state.minprice > 100 ? 'true' : ''}>100</option>
                                                <option value="200" disabled={this.state.minprice > 200 ? 'true' : ''}>200</option>
                                                <option value="300" disabled={this.state.minprice > 300 ? 'true' : ''}>300</option>
                                                <option value="500" disabled={this.state.minprice > 500 ? 'true' : ''}>500</option>
                                                <option value="1000">1000</option>
                                            </select>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-md-12">
                                    <input type="submit" name="submit" id="submit" value="Find Room" className="btn btn-primary btn-block"></input>
                                </div>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        )
    }   
}

export default FormSearch;

