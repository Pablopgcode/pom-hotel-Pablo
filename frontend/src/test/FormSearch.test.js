import React from 'react';
import FormSearch from 'components/Forms/FormSearch';
import {shallow} from 'enzyme';
import {now} from '../services/dateservice';

describe('Search form unit test', () => {
    test('it should render without crash', () => {
        const wrapper = shallow(<FormSearch />);
        expect(wrapper).toMatchSnapshot(); 
    });

    test('should have required inputs [',() => {
        const wrapper = shallow(<FormSearch />);
        expect(wrapper.find('[name="guests"]')).toHaveLength(1);
    });

    test('should run onChange with the new state', () => {
        const updateFilter = jest.fn();
        const wrapper = shallow(<FormSearch onFilterChange={updateFilter}/>);
        wrapper.find('select[name="guests"]').simulate('change', {target:{name:'guests', value: '3'}});
        wrapper.find('form').simulate('submit', {preventDefault: () => {}});


        expect (updateFilter).toHaveBeenCalledWith({
                rtypes: [],
                guests: "3",
                startDate: now,
                endDate: now,
                type: "",
                minprice: "1",
                maxprice: "1000"
        })
    });

});