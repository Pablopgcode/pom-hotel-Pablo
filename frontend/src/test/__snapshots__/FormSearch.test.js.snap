// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search form unit test it should render without crash 1`] = `ShallowWrapper {}`;
